### ISSUE TITLE
[BUG] replacement for npm_config_argv in v8.5.0

### ISSUE THREAD

**Issue by HesterG** [npm/cli#4663]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

When trying to use `npm_config_argv`, it appears to be removed. I am wondering what is the replacement for `npm_config_argv`, i tried to search around, and it says "Where the argv is being parsed in order to determine the command being run, the npm_command environ provides a safer approach." However, the `npm_command` currently holds a value of `exec`, does it mean this is not yet implemented?

### Steps To Reproduce

```
console.log(process.env.npm_config_argv)
```
`process.env.npm_config_argv` is `undefined`

### Environment

- npm: `8.5.0`
- Node.js: `v16.14.2`
- OS Name: `MacOS 10.15.7`

**Comment by nlf** [comment]
`npm_config_argv` was one of the things that was intentionally removed in npm v7. we no longer export the arguments that were provided to npm itself in an environment variable anywhere, and at this time we do not plan to.

i'm curious, what's your use case? it's possible there are other ways to get the data you're looking for, but i'd have to know exactly what you're trying to get from it.

**Comment by peti29** [comment]
I also miss that feature.
I would need it in a postbuild script to do some cleanup after the build script was run.

I have an Angular project that has several libs. In the past I specified which lib to build like this:
ng build ngx-mycomponent-a --configuration development

However now in one of our components we need to compile a couple of scss files to stand alone css. We also want to put these into an assets/themes folder under dist/.

Unfortunately the angular build options for styles does not allow to set an ouput folder. The resulting css files will end up directly under dist/. (I'm talking about the inject: false and bundleName: someName style options)

Thus I created a postbuild script that will move the css files from under dist/ and put them under dist/assets/themes.
Now we can no longer use ng build. We need to call:
npm run build ngx-mycomponent-a -- --configuration development

But I found that an ugly solution if the css-s from ngx-mycomponent-a will be moved even if I'm building ngx-mycomponent-b. So I came up with the idea of only performing that action when ngx-mycomponent-a is being built.

I can do it relying on $env:npm_config_argv in my postbuild script.

However that only works under NPM v6.* because there is no npm_config_argv in NPM v7+.

I have found NO solution to this. How could I tell from my postbuild script which component is being built in the build script???

(I'm on Windows)

**Comment by HesterG** [comment]
> `npm_config_argv` was one of the things that was intentionally removed in npm v7. we no longer export the arguments that were provided to npm itself in an environment variable anywhere, and at this time we do not plan to.
> 
> i'm curious, what's your use case? it's possible there are other ways to get the data you're looking for, but i'd have to know exactly what you're trying to get from it.

Yes i used another way to get the params by doing something like `npm run <command> -- --param=value `, and then use `process.argv` to get the param value.

**Comment by suxianbaozi** [comment]
> `npm_config_argv` was one of the things that was intentionally removed in npm v7. we no longer export the arguments that were provided to npm itself in an environment variable anywhere, and at this time we do not plan to.
> 
> i'm curious, what's your use case? it's possible there are other ways to get the data you're looking for, but i'd have to know exactly what you're trying to get from it.

I have a project , it was a webpack plugin , 
when run   " npm run test --test 1,2,3. " which will publish the build result to the test evironment 1,2,3

in this plugin , i have to got this value

**Comment by tip0x** [comment]
If npm_config_argv has been removed, what is the replacement?
here is my code `const argv = JSON.parse(env("npm_config_argv"));
if (argv.original !== [ "Hardhat", "test "]) {
  require('dotenv').config();
}
`
it returns "An unexpected error occurred.
npm_config_argv is undefined" when I run npx hardhat compile

**Comment by samuelneff** [comment]
I have a `preinstall` script to verify that `--no-recurse` was passed to install and fail install if not. I don't see how to do this without the npm command line parameters available; my goal here is specifically to validate them.

Perhaps there's another way to accomplish the same thing?
